be/src/exprs/dict_query_expr.cpp

```diff
@@ -0,0 +1,140 @@
+        columns[i] = _children[i]->evaluate(context, ptr);
+    }
+
+    ColumnPtr res;
+    for (auto& column : columns) {
+        if (column->is_constant()) {
+            column = ColumnHelper::unpack_and_duplicate_const_column(size, column);
+        }
+    }
+    ChunkPtr key_chunk = ChunkHelper::new_chunk(_key_slots, size);
+    key_chunk->reset();
+    for (int i = 0; i < _dict_query_expr.key_fields.size(); ++i) {
+        ColumnPtr key_column = columns[1 + i];
+        key_chunk->update_column_by_index(key_column, i);
+    }
+
+    for (auto& column : key_chunk->columns()) {
+        if (column->is_nullable()) {
+            column = ColumnHelper::update_column_nullable(false, column, column->size());
+        }
+    }
+
+    std::vector<bool> found;
+    ChunkPtr value_chunk = ChunkHelper::new_chunk(_value_slot, key_chunk->num_rows());
+
+    Status status = _table_reader->multi_get(*key_chunk, {_dict_query_expr.value_field}, found, *value_chunk);
+    if (!status.ok()) {
+        // todo retry
+        LOG(WARNING) << "fail to execute multi get: " << status.detailed_message();
+        return status;
+    }
+    res = ColumnHelper::create_column(_value_slot[0]->type(), true);
+
+    int res_idx = 0;
+    for (int idx = 0; idx < size; ++idx) {
+        if (found[idx]) {
+            res->append_datum(value_chunk->get_column_by_index(0)->get(res_idx));
+            res_idx++;
+        } else {
+            if (_dict_query_expr.strict_mode) {
+                return Status::NotFound("In strict mode, query failed if record not exist in dict table.");
+            }
+            res->append_nulls(1);
+        }
+    }
+
+    return res;
+}
+
+Status DictQueryExpr::prepare(RuntimeState* state, ExprContext* context) {
+    RETURN_IF_ERROR(Expr::prepare(state, context));
+    _runtime_state = state;
+    return Status::OK();
+}
+
+Status DictQueryExpr::open(RuntimeState* state, ExprContext* context, FunctionContext::FunctionStateScope scope) {
+    // init parent open
+    RETURN_IF_ERROR(Expr::open(state, context, scope));
+
+    TGetDictQueryParamRequest request;
+    request.__set_db_name(_dict_query_expr.db_name);
+    request.__set_table_name(_dict_query_expr.tbl_name);
+    TGetDictQueryParamResponse response;
+
+    TNetworkAddress master_addr = get_master_address();
+    RETURN_IF_ERROR(ThriftRpcHelper::rpc<FrontendServiceClient>(
+            master_addr.hostname, master_addr.port,
+            [&request, &response](FrontendServiceConnection& client) { client->getDictQueryParam(response, request); },
+            30000));
+
+    TableReaderParams params;
+    params.schema = response.schema;
+    params.partition_param = response.partition;
+    params.location_param = response.location;
+    params.nodes_info = response.nodes_info;
+    params.partition_versions = _dict_query_expr.partition_version;
+    params.timeout_ms = 30000;
+
+    _table_reader = std::make_shared<TableReader>();
+    _table_reader->init(params);
```

Comment:
check return status from init.

Reply:
Expr::open will be called in pipeline::prepare. these operator may block pipeline executor.

Reply:
It seems just some params set, won't block main thread.